cpg_workflows/stages/seqr_loader_long_read/long_read_sv_annotation.py

```diff
@@ -53,20 +72,115 @@ def query_for_sv_vcfs(dataset_name: str) -> dict[str, str]:
-                return_dict[sg['id']] = analysis['output']
+            if not analysis['meta'].get('variant_type') == 'SV':
+                continue
+            if analysis['meta'].get('joint_called', False):
+                joint_called_vcfs[sg['id']] = {
+                    'output': analysis['output'],
+                    'meta': analysis['meta'],
+                }
+            else:
+                single_sample_vcfs[sg['id']] = {
+                    'output': analysis['output'],
+                    'meta': analysis['meta'],
+                }
+
+    # Prefer the joint-called VCFs over the single-sample VCFs
+    sg_vcfs = {}
+    for sg_id, single_sample_vcf in single_sample_vcfs.items():
+        if sg_id not in joint_called_vcfs:
+            sg_vcfs[sg_id] = single_sample_vcf
+            continue
+        sg_vcfs[sg_id] = joint_called_vcfs[sg_id]
+
+    # Remove the parents entries if their family has a joint-called VCF
+    sgs_to_skip = find_sgs_to_skip(sg_vcfs)
+    return_dict = {}
+    for sg_id, vcf_analysis in sg_vcfs.items():
+        if sg_id in sgs_to_skip:
+            get_logger().info(f'Skipping {sg_id} as it is a parent in a joint-called VCF')
+            continue
+        return_dict[sg_id] = vcf_analysis
 
     return return_dict
 
 
+def find_sgs_to_skip(sg_vcf_dict: dict[str, dict]) -> set[str]:
+    """
+    Find the SGs to skip in the reformatting stage
+    These are the parents if the family has been joint-called
+    """
+    sgs_to_skip = set()
+    joint_called_families = set()
+    for sg_id, vcf_analysis in sg_vcf_dict.items():
+        analysis_meta = vcf_analysis['meta']
+        if analysis_meta.get('joint_called', False):
+            joint_called_families.add(analysis_meta.get('family_id', ''))
+    for sg_id, vcf_analysis in sg_vcf_dict.items():
+        analysis_meta = vcf_analysis['meta']
+        if analysis_meta.get('family_id', '') in joint_called_families and not analysis_meta.get('joint_called', False):
+            sgs_to_skip.add(sg_id)
+    return sgs_to_skip
+
+
+def query_for_lrs_sg_id_mapping(datasets: list[str]):
+    """
+    Query metamist for the LRS ID corresponding to each sequencing group ID
+    """
+    lrs_sgid_mapping = {}
+    for dataset in datasets:
+        if config_retrieve(['workflow', 'access_level']) == 'test' and not dataset.endswith('-test'):
+            dataset = dataset + '-test'
+        query_results = query(LRS_IDS_QUERY, variables={'dataset': dataset})
+        for sg in query_results['project']['sequencingGroups']:
+            sample = sg['sample']
+            participant = sample['participant']
+            lrs_id = sample['meta'].get('lrs_id', None)
+            if not lrs_id:
+                get_logger().warning(
+                    f'{dataset} :: No LRS ID found for {participant["externalId"]} - {sample["externalId"]}',
+                )
+                continue
+            lrs_sgid_mapping[lrs_id] = sg['id']
+    return lrs_sgid_mapping
```

Comment:
you can import both these methods from the other script, or pull them out into a `lr_utils.py`. Not a big deal, but would make it easier if anything changes in future